userspace/libsinsp/cri.cpp

```diff
@@ -301,10 +301,24 @@ bool cri_interface::parse_cri_runtime_spec(const Json::Value &info, sinsp_contai
 		set_numeric_32(*cpu, "cpuset_cpu_count", container.m_cpuset_cpu_count);
 	}
 
+	bool priv_found = false;
 	const Json::Value *privileged;
 	if(walk_down_json(*linux, &privileged, "security_context", "privileged") && privileged->isBool())
 	{
 		container.m_privileged = privileged->asBool();
+		priv_found = true;
+	}
+
+	// containerd
```

Comment:
The chunk above is also expected to work for containerd. Did something change between versions?

Reply:
I think it's possible that it was changed at some point...in `cri.proto` the Info field is defined as
```
Info                 map[string]string `protobuf:"bytes,2,rep,name=info,proto3" json:"info,omitempty" protobuf_key:"bytes,1,opt,name=key,proto3" protobuf_val:"bytes,2,opt,name=value,proto3"`
```
Since it doesn't say where those flags should be located, container runtime can create backward-incompatible change accidentally.  

I wonder which version of containerd was tested previously?  I can take a look to see when this was changed.

Reply:
Hmm, hard to say but it's been a while ago:

```
commit 3e24f84696631b7fec23ce929795e328722f4946
Author: Grzegorz Nosek <grzegorz.nosek@sysdig.com>
Date:   Mon Dec 17 19:53:23 2018 +0100

    Initial CRI support
```

The logic hasn't changed since, even though it's been moved around quite a bit.

I want to say 1.1.something but, well, it's been a few years now :)

Reply:
Hmm I couldn't find any significant changes in `containerd` and `runtime-spec` since 2018.  The [runtimeSpec->Linux](https://github.com/opencontainers/runtime-spec/blob/bf1ee9178eaa1186ed1fb670a50f02e9d3928bb5/specs-go/config.go#L146) doesn't have `security_context` anymore too as far as I can tell.

@gnosek What's your opinion about this?  I'm ok with either keeping or removing the existing containerd logic.

Reply:
I'd say leave it, with an `// old containerd?` comment to match your newer ones